packages/experimental/Shimmer/src/Shimmer.win32.tsx

```diff
@@ -0,0 +1,100 @@
+
+  return (
+    <Svg {...props} width="100%" height="100%">
+      <LinearGradient id="gradient">
+        <Stop stopColor={shimmerColor} />
+        <Stop offset="20%" stopColor={shimmerWaveColor} />
+        <Stop offset="40%" stopColor={shimmerColor} />
+      </LinearGradient>
+      <Rect width={props.shimmerWaveWidth} height="100%" fill="url(#gradient)" fillOpacity={props.gradientOpacity} />
+    </Svg>
+  );
+};
+
+const waveContainer: React.FunctionComponent<ShimmerTokens> = (props: ShimmerTokens) => {
+  return <RCTNativeAnimatedShimmer {...{ ...props, style: { backgroundColor: props.shimmerColor, overflow: 'hidden' } }} />;
+};
+
+export const Shimmer = compose<ShimmerType>({
+  displayName: shimmerName,
+  ...styleSettings,
+
+  slots: {
+    root: View,
+    clippingMask: clippingMask,
+    shimmerWave: wave,
+    shimmerWaveContainer: waveContainer,
+  },
+
+  render: (props: ShimmerProps, useSlots: UseSlots<ShimmerType>) => {
+    return (rest: ShimmerProps) => {
+      const { elements, ...mergedProps } = mergeProps(props, rest);
+      const Slots = useSlots(mergedProps);
+
+      /**
+       * Win32 D2D1 doesn't directly support SVG mask elements, so we're going to generate a mask
+       * by authoring a clip path for the same effect.  We do so utilizing the clip-rule='evenodd'
+       * and drawing a rectangle around the entire svg viewbox, causing all inner elements to be "clipped out"
+       * instead of "clipped around," generating the visual we desire with the linear gradient below the mask.
+       *
+       * The layout properties and elements we've been given, however, may not be known at render time.  Rather
+       * than try to keep up with the actual size of our control, we'll draw a very large rectangle well outside the range
+       * of our generated coordinates.  This large rectangle will ensure we keep the desired 'evenodd' clipping behavior
+       * without risking intersection with the provided Shimmer elements.
+       */
+      let clipPathsAsMask: string[] = ['M -1 -1 h 100000 v 100000 h -100000 z '];
+
+      /**
+       * Now extend the path to include the provided element shapes, with their path parameter strings generated from their corresponding properties.
+       * Ideally this would be better possible with functions from the shape elements themselves and not to rely on our own calculations.
+       */
+
+      if (elements) {
+        for (let i = 0; i < elements.length; i++) {
+          const element = elements[i];
+          clipPathsAsMask = clipPathsAsMask.concat(
+            element.type === 'circle' ? convertCircleToSvgPath(element) : convertRectToSvgPath(element),
+          );
+        }
+      }
+
+      return (
+        <Slots.root {...mergedProps}>
+          <Slots.shimmerWaveContainer>
+            <Slots.shimmerWave />
+          </Slots.shimmerWaveContainer>
+          <Slots.clippingMask clipPath={clipPathsAsMask.join(' ')} />
```

Comment:
could we pull out the join to it's own var outside of the return?

Reply:
what's the gain?

Reply:
Forgot to prefix my comment with nit.
Mostly keeping logic outside of the JSX. At one point Jason told me not to pass subobjects inside JSX as it forces js to recreate the object on every render. I'm not sure that optimization applies here.